### PR TITLE
persist time to tablet in bulk update

### PR DESCRIPTION
**Pull request by keith-turner**
When bulk import operations set time and a tablet was hosted the time was not persisted.  The bulk import fate operation now persist time in tablet metadata.  The tablet code assumed it was the only thing updating a tablets time field.  The tablet code was modified to accomodate the bulk import code running in the manager updating the tablets time column in the metadata table.

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 1]
```diff
@@ -1302,63 +1304,77 @@ public Optional<StoredTabletFile> updateTabletDataFile(long maxCommittedTime,
...
+          tablet.putFile(newDatafile, dfv);
+          newFile = Optional.of(newDatafile.insert());
+        }
 
-      tablet.putFlushId(flushId);
+        boolean setTime = false;
+        // bulk imports can also update time in the metadata table, so only update if we are moving
+        // time forward
+        if (maxCommittedTime > lastTabletMetadata.getTime().getTime()) {
+          // TODO throw exception if requireSame called more than once
```
Comment (dlmarion) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java:1329]: Should this `TODO` be an `ELASTICITY_TODO`?

**Review by dlmarion — COMMENTED** [review]
I'm not quite sure how tablet time works in general. It might be useful to have a discussion about it so that I/we can understand the changes here.

**Comment by keith-turner** [comment]
> I'm not quite sure how tablet time works in general.

Here is some info.  Each tablet has a concept of time that only moves forward.  This time is persisted in the tablet metadata table entry.  As mutations arrive in a tablet, if the time was not explicitly set then its set on the mutation and tablets time is incremented.  For bulk import one can optionally set the tablets current time on an entire file.  The way this works is that it allocates a timestamp for the tablet and then persist this with the bulk file entry.  When the bulk file is read, if a timestamp is present for the file then its applied to everything in the file.  

 1. Batch write row=a value=1
 2. Bulk import row=a value=2 and set time.  

With the ability to set time on bulk imports, it allows bulk imports to be properly orders with write.  Like the above would set the timestamp on the 2nd write by the bulk import such that its higher than the timestamp on the first write.

This PR adds coordination between a hosted tablet and the bulk import code running in the manager to ensure that timestamp is set correctly.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java (hunk 1 above)
Comment (keith-turner) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: I looked in the comment and found my assumption was wrong and the comment was not needed, so removed it.

**Comment by dlmarion** [comment]
Does this close #3354 ?

**Pull request merged**